iothub_client/tests/common_longhaul/iothub_client_common_longhaul.c

```diff
@@ -1675,11 +1675,20 @@ static void on_twin_report_state_completed(int status_code, void* userContextCal
 
 static void check_for_reported_properties_update_on_service_side(IOTHUB_LONGHAUL_RESOURCES* iotHubLonghaul)
 {
-    char* twin;
+    char* twin = NULL;
 
-    if ((twin = IoTHubDeviceTwin_GetTwin(
-        iotHubLonghaul->iotHubSvcDevTwinHandle,
-        iotHubLonghaul->deviceInfo->deviceId)) == NULL)
+    for (int i = 0; i < 3 && twin == NULL; i++)
+    {
+        twin = IoTHubDeviceTwin_GetTwin(iotHubLonghaul->iotHubSvcDevTwinHandle, iotHubLonghaul->deviceInfo->deviceId);
+        if (twin == NULL)
+        {
+            // this API can fail due to network issues (ex: DNS issues)
+            // try a few times before reporting an error
```

Comment:
Is that documented somewhere online?

Reply:
no, empirical determined from the callstack trace. I suspect all service APIs do this.